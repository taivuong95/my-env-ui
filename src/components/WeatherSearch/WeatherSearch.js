import React, { useContext } from 'react';
import Context from '../../contexts/Context'
import Loader from '../UI/Loader';

const WeatherSearch = () => {
    const { getWeatherInfo, city, loading } = useContext(Context)
    return (
        <div data-test="weather-search" className="weather-search">
            {loading ? <Loader /> : <div className="weather-search--city">{city || 'Please enter the name of the city to get started!'}</div>}
            <form data-test="weather-search__form" onSubmit={getWeatherInfo} className="weather-search__form" >
                <input data-test="city"  name="location" placeholder="Type the city" autoComplete="off" className="weather-search__input" type="text" />
                <div className="weather-search__submit">
                    <button className="weather-search__button">&rarr;</button>
                </div>
            </form>
        </div>
    )
}

export default WeatherSearch