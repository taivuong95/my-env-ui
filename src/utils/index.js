export const KelvinToCelsius = (kelvin) => {
    return kelvin - 273.15;
}