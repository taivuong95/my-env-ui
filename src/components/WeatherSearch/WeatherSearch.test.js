import React from 'react';
import WeatherSearch from './WeatherSearch';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { findByTestAttr } from '../../utils/index';
import Context from '../../contexts/Context';
import Loader from '../UI/Loader';

const value = {
    getWeatherInfo: jest.fn(),
    city: 'Paris',
    loading: true
}

describe('WeatherSearch Component', () => {
    test("expect to render WeatherSearch component", () => {
        const wrapper = mount(<Context.Provider value={value}><WeatherSearch /> </Context.Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('renders without error', () => {
        const wrapper = mount(<Context.Provider value={value}><WeatherSearch /> </Context.Provider>)
        const weatherSearchComponent = findByTestAttr(wrapper, "weather-search");
        expect(weatherSearchComponent.length).toBe(1);
    });
    test(" render a loading before api call success(loading is true)", () => {
        const wrapper = mount(<Context.Provider value={value}><Loader /> </Context.Provider>)
        expect(wrapper.find(".loader").exists()).toBeTruthy();
    });

})