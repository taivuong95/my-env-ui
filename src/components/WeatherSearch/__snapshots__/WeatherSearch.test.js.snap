// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherSearch Component expect to render WeatherSearch component 1`] = `
Array [
  <WeatherSearch>
    <div
      className="weather-search"
      data-test="weather-search"
    >
      <Loader>
        <div
          className="loader"
        />
      </Loader>
      <form
        className="weather-search__form"
        data-test="weather-search__form"
        onSubmit={[MockFunction]}
      >
        <input
          autoComplete="off"
          className="weather-search__input"
          data-test="city"
          name="location"
          placeholder="Type the city"
          type="text"
        />
        <div
          className="weather-search__submit"
        >
          <button
            className="weather-search__button"
          >
            â†’
          </button>
        </div>
      </form>
    </div>
  </WeatherSearch>,
  " ",
]
`;
